test/fw/ptl/utils/pbs_testsuite.py

```diff
@@ -1168,33 +1168,80 @@ def _revert_pbsconf_server(self, vals_to_set):
+                                self.logger.error('comm '
+                                                  + self.comm.hostname
+                                                  + ' is down')
+                                msg = 'Failed to restart comm '
+                                msg += self.comm.hostname
+                                self.assertTrue(rv, msg)
                 else:
                     for initcmd in cmds_to_exec:
                         # start/stop the particular daemon
                         server.pi.initd(server.hostname, initcmd[1],
                                         daemon=initcmd[0])
                         if initcmd[1] == "start":
                             if initcmd[0] == "server":
-                                if not server.isUp():
-                                    self.fail("Server is not up")
-                            if initcmd[1] == "sched":
-                                if not self.sched.isUp():
-                                    self.fail("Scheduler is not up")
+                                rv = server.isUp()
+                                if not rv:
+                                    self.logger.error('Server '
+                                                      + comm.hostname
+                                                      + ' is down')
+                                    msg = 'Failed to restart server '
+                                    msg += server.hostname
+                                    self.assertTrue(rv, msg)
+                            if initcmd[0] == "sched":
+                                rv = server.schedulers['default'].isUp()
+                                if not rv:
+                                    self.logger.error('Scheduler is down')
+                                    msg = 'Failed to restart scheduler '
+                                    msg += server.scheduler.hostname
+                                    self.assertTrue(rv, msg)
                             if initcmd[0] == "mom":
-                                if not self.mom.isUp():
-                                    self.fail("Mom is not up")
+                                rv = server.moms.values()[0].isUp()
+                                if not rv:
+                                        self.logger.error('mom '
+                                                          + server.mom.hostname
+                                                          + ' is down')
+                                        msg = 'Failed to restart mom '
+                                        msg += server.mom.hostname
+                                        self.assertTrue(rv, msg)
                             if initcmd[0] == "comm":
-                                if not self.comm.isUp():
-                                    self.fail("comm is not running")
+                                if self.du.is_localhost(server.hostname):
```

Comment:
This is up to you, but i feel like we could do some refactoring here to reuse a lot of the code. Suggestion: create a function called _check_daemon_on_server(server_obj, daemon_name), which will check the daemon name and call isUp on sched, mom or comm and print the error message as relevant. I think this method will shrink down a lot if we do that.